packages/block-library/src/quote/v2/edit.js

```diff
@@ -9,24 +9,66 @@ import {
 	store as blockEditorStore,
 } from '@wordpress/block-editor';
 import { BlockQuotation } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useDispatch, useSelect, useRegistry } from '@wordpress/data';
 import { createBlock } from '@wordpress/blocks';
-import { Platform } from '@wordpress/element';
+import { Platform, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { migrateToQuoteV2 } from './deprecated';
 
 const isWebPlatform = Platform.OS === 'web';
 const TEMPLATE = [ [ 'core/paragraph', {} ] ];
 
+/**
+ * At the moment, deprecations don't handle create blocks from attributes
+ * (like when using CPT templates). For this reason, this hook is necessary
+ * to avoid breaking templates using the old quote block format.
+ *
+ * @param {Object} attributes Block attributes.
+ * @param {string} clientId   Block client ID.
+ */
+const useMigrateOnLoad = ( attributes, clientId ) => {
+	const registry = useRegistry();
+	const { updateBlockAttributes, replaceInnerBlocks } = useDispatch(
+		blockEditorStore
+	);
+	useEffect( () => {
+		// As soon as the block is loaded, migrate it to the new version.
+
+		if ( ! attributes.value ) {
+			// No need to migrate if it doesn't have the value attribute.
+			return;
+		}
+
+		const [ newAttributes, newInnerBlocks ] = migrateToQuoteV2(
+			attributes
+		);
```

Comment:
Any thoughts on adding a `deprecated` call here. I considered adding one for the list block too.

Reply:
To inform the user that some of the templates they are using need updating? Sounds sensible.

What about ?

```suggestion
		);
		
		deprecated( 'Do not use the value attribute on the quote block', {
			since: '12.10', // To be updated when v2 is the default
			alternative: 'Use inner blocks instead'
		});
```

Reply:
```suggestion
		);

		deprecated( 'Value attribute on the quote block', {
			since: '6.0', // WP version
			version: '6.5' // when the attribute gets removed. We can drop this property to never remove the deprecation
			alternative: 'inner blocks'
		});
```